modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/DiscountedCumulativeGainAt.java

```diff
@@ -178,19 +193,50 @@ private static double computeDCG(List<Integer> ratings) {
         PARSER.declareInt(DiscountedCumulativeGainAt::setUnknownDocRating, UNKNOWN_DOC_RATING_FIELD);
     }
 
+    @Override
+    public DiscountedCumulativeGainAt fromXContent(XContentParser parser, ParseFieldMatcher matcher) {
+        return DiscountedCumulativeGainAt.fromXContent(parser, new ParseFieldMatcherSupplier() {
+            @Override
+            public ParseFieldMatcher getParseFieldMatcher() {
+                return matcher;
+            }
+        });
+    }
+
     public static DiscountedCumulativeGainAt fromXContent(XContentParser parser, ParseFieldMatcherSupplier matcher) {
         return PARSER.apply(parser, matcher);
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
         builder.startObject(NAME);
         builder.field(SIZE_FIELD.getPreferredName(), this.position);
         builder.field(NORMALIZE_FIELD.getPreferredName(), this.normalize);
         if (unknownDocRating != null) {
             builder.field(UNKNOWN_DOC_RATING_FIELD.getPreferredName(), this.unknownDocRating);
         }
         builder.endObject();
+        builder.endObject();
         return builder;
     }
+    
+    @Override
+    public final boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        DiscountedCumulativeGainAt other = (DiscountedCumulativeGainAt) obj;
+        return Objects.equals(position, other.position) &&
+                Objects.equals(normalize, other.normalize) &&
+                Objects.equals(unknownDocRating, other.unknownDocRating);
+    }
+    
+    @Override
+    public final int hashCode() {
+        return Objects.hash(getClass(), position, normalize, unknownDocRating);
```

Comment:
Just stumbled across this, I see we also have getClass() in the hashCode of e.g. AbstractQueryBuilder but I can't remember why this was necessary. I did some research for best practices but couldn't find anything yet, maybe you have some pros / cons? Only asking because I would like to understand if or why it is necessary.